src/Model_Hydro/CPU_Hydro/CPU_Shared_FullStepUpdate.cpp

```diff
@@ -164,15 +166,37 @@ void Hydro_FullStepUpdate( const real g_Input[][ CUBE(FLU_NXT) ], real g_Output[
       for (int v=0; v<NCOMP_TOTAL; v++)   g_Output[v][idx_out] = Output_1Cell[v];
 
 
-//    5. check the negative density and energy
-#     ifdef CHECK_NEGATIVE_IN_FLUID
-      if ( Hydro_CheckNegative(Output_1Cell[DENS]) )
-         printf( "WARNING : invalid density (%14.7e) at file <%s>, line <%d>, function <%s>\n",
-                 Output_1Cell[DENS], __FILE__, __LINE__, __FUNCTION__ );
+//    5. check unphysical cells within a patch
 
-      if ( Hydro_CheckNegative(Output_1Cell[ENGY]) )
-         printf( "WARNING : invalid energy (%14.7e) at file <%s>, line <%d>, function <%s>\n",
-                 Output_1Cell[ENGY], __FILE__, __LINE__, __FUNCTION__ );
+      if ( State != NULL )
+      {
+
+        if(Hydro_CheckUnphysical( Output_1Cell, NULL, NULL, NULL, NULL, __FILE__, __FUNCTION__, __LINE__, false ))
+        {
+#         ifdef __CUDACC__
+          atomicOr ( (int*)State, 1);
```

Comment:
- [ ] It seems to me that we only need the *block-wide* atomic function `atomicOr_block`. See the [CUDA manual](https://docs.nvidia.com/cuda/cuda-c-programming-guide/index.html#atomic-functions). But I'm not very familiar with atomic functions so please test it carefully.
- [ ] Can it be simplified as `atomicExch( State, 1 );`?
- [ ] Use three spaces for indentation.